### ISSUE TITLE
should oce.plot.ts(type="l") connect across NA points?

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2254]
In fixing #2241, I noticed that `oce.plot.ts(..., type = "l")`  will continue the curve across missing-data points.  But R doesn't do that for `plot()`, so I consider this to be erroneous behaviour in `oce.plot.ts().

The code and screenshot below illustrates.

Since `oce.plot.ts()` is an old function that has been used a *lot*, I am a bit reluctant to make a change without discussing this with the co-developers, @richardsc and @clayton33.

I don't think there is a huge rush on this.  I'm a +0.5 vote on changing `oce.plot.ts()`.  But maybe, to get consistency with past behaviour, it needs a new argument (named something like `spanNA` or similar word ... naming is always hard)

```R
library(oce)
N <- 50000 # make long to force oce.plot.ts() to simplify it
y <- sin(seq_len(N) * 2 * pi / 5000)
t <- as.POSIXct(Sys.Date()) + 1:N
y[1000:5000] <- NA
par(mfrow=c(2, 1))
oce.plot.ts(t, y, type="p", cex = 0.2)
oce.plot.ts(t, y, type="l")
```
<img width="752" alt="Screenshot 2024-10-06 at 10 17 53 AM" src="https://github.com/user-attachments/assets/c519897e-2add-452f-a2f2-2823d0e00a8e">

**Comment by dankelley** [comment]
Oh, this is **only** for the case where the data have been "simplified", as shown below.

That is:
* for large datasets the gaps are filled in
* but for small datasets, gaps are not filled in

so this is clearly a bug, and there's no need for voting before deciding whether to fix it.  I'll have a look today, but if not then, maybe during the week.  Again, this only happens for very large datasets, and it only happens if a particular sub-interval has *no* good data.  And the behaviour is *clearly* erroneous, because it disagrees with base `plot()` and also with `oce.plot.ts()` with either smaller datasets or even large ones, so long as there is at least some data in each subinterval.

I realize that my wording here is a bit confusing, but sometimes figuring out how to exactly phrase a bug report can be slower than finding the problem.

``` r
library(oce)
#> Loading required package: gsw
N <- 500 # make long to force oce.plot.ts() to simplify it
y <- sin(seq_len(N) * 2 * pi / 50)
t <- as.POSIXct(Sys.Date()) + 1:N
y[100:200] <- NA
par(mfcol=c(2, 2))
oce.plot.ts(t, y, type="p", cex = 0.2)
oce.plot.ts(t, y, type="l")
X <- 1:10
Y <- X * (1 - X)
Y[3:5] <- NA
plot(X, Y)
plot(X, Y, type = "l")
```

![](https://i.imgur.com/72Hp79j.png)

<sup>Created on 2024-10-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by dankelley** [comment]
This is fixed in commit 454cd50a5c6878bfcec5b7c13921a03a707929b7 of he "develop" branch.

The oce-issues repo has an example (named for this issue number).  A reprex is below.

``` r
# Large dataset: problem with filling across NAs
library(oce)
#> Loading required package: gsw
N <- 50000 # make long to force oce.plot.ts() to simplify it
y <- sin(seq_len(N) * 2 * pi / 5000)
t <- as.POSIXct(Sys.Date()) + 1:N
y[1000:5000] <- NA
par(mfcol = c(2, 2))
oce.plot.ts(t, y, type = "p", cex = 0.2)
oce.plot.ts(t, y, type = "l")
#> Warning in oce.plot.ts(t, y, type = "l"): simplifying a large dataset; set simplify=NA to see raw data
X <- 1:10
Y <- X * (1 - X)
Y[3:5] <- NA
plot(X, Y)
plot(X, Y, type = "l")
```

![](https://i.imgur.com/SsnlCwp.png)

<sup>Created on 2024-10-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Below is a reprex.
